### DISCUSSION TITLE
A setting to make courseplay actually stay inside the field boundary?

### DISCUSSION THREAD

**Discussion by cyr123** [Courseplay/Courseplay_FS22#2613] [Ideas]
I'm not actually sure if this is a bug or a new feature request, but I expect it's the latter.

My biggest frustration with courseplay is trying to keep it from getting stuck on trees/fences/telephone poles or whatever else is near the edge of the field. There is the setting to only turn in the field, but even when that seems to work it would appear that it only keeps the actual path of the turn (the centerline of the vehicle) inside the boundary - which means there could be parts (mowers, headers, sprayer booms) sticking out 5, 10 or 20 meters outside!

One way to handle this, at least on fields where there is enough space for multiple headlands, would be a "turn inside first headland" setting, but better IMHO would be if CP could automatically detect the maximum "stickout" of the vehicle/implements and keep the turns away from the boundary by at least that distance.

**Comment by Tensuko** [comment]
Turn on headland must be on and you need enough headlands.
Also try out pathfinder turn.

**Comment by guerrillian** [comment]
Here are some tips that worked for me, the work is slower but I like it:

1 - Draw a custom field and exceed the field margin by moving the points, you'll use more resources like seed, but it is what it is
2 - Do **sharp** headlands, sometimes 1 headland is enough, you need to see the course it is generating to know if it'll turn off the field 
3 - Set Late tools to force the tractor to use reverse at turns

Custom field
![image](https://github.com/Courseplay/Courseplay_FS22/assets/128705907/44c51104-83cb-4130-9876-e1b8e43be0d1)

Late tools
![image](https://github.com/Courseplay/Courseplay_FS22/assets/128705907/b24d2e85-5002-4bbf-84f9-34fc0c6392eb)

Sharp headland and standard tools vs Sharp headland and later tools
![image](https://github.com/Courseplay/Courseplay_FS22/assets/128705907/227e28f0-61a0-4014-8d41-aa94e81ca12e)

Exceeding the field
![image](https://github.com/Courseplay/Courseplay_FS22/assets/128705907/98c3a0b3-68a2-4e82-9aae-6d66e7187ba0)

**Comment by coatsy35** [comment]
> I'm not actually sure if this is a bug or a new feature request, but I expect it's the latter.
> 
> My biggest frustration with courseplay is trying to keep it from getting stuck on trees/fences/telephone poles or whatever else is near the edge of the field. There is the setting to only turn in the field, but even when that seems to work it would appear that it only keeps the actual path of the turn (the centerline of the vehicle) inside the boundary - which means there could be parts (mowers, headers, sprayer booms) sticking out 5, 10 or 20 meters outside!
> 
> One way to handle this, at least on fields where there is enough space for multiple headlands, would be a "turn inside first headland" setting, but better IMHO would be if CP could automatically detect the maximum "stickout" of the vehicle/implements and keep the turns away from the boundary by at least that distance.

It would help if CP told us the length of the vehicle and implement combination and what it's recommended number of headland lines is. I too would love to see a feature that keeps it inside the outer field boundary. I often see it pull into a shaft corner and try to "nose" out of the field past the outer boundary and then get stuck as it' can't pull far enough forward to finish it's run. It should shop before the outer edge of the field and the nrevers back as the part now underneath the implement was probably covered on the last run anyway so doesn't need to draw that far forward. IRL current systems that can perform automatic headland turns use the fields outer boundary as a hard line and make sure any path created taking into account the vehicle width turns inside the boundary otherwise you would be driving on top of a hedge or sticking a sprayer book into the trees etc....
